lua/apisix/core/config_etcd.lua

```diff
@@ -194,9 +210,13 @@ local function sync_data(self)
     log.debug("res: ", json.delay_encode(res, true))
     log.debug("headers: ", json.delay_encode(headers, true))
     if not res then
-        if headers then
-          self:upgrade_version(headers["X-Etcd-Index"])
+        if err == "The event in requested index is outdated and cleared" then
```

Comment:
It will fail when etcd changed this error msg. Is there a better way?

Reply:
We can check the error messages returned for different etcd versions.

Reply:
So we need a test case to cover it, otherwise, we can not catch the changes.

Reply:
It's not easy to test it in APISIX. We can confirm it in lua-resty-etcd.

Reply:
https://github.com/etcd-io/etcd/blob/a621d807f061e1dd635033a8d6bc261461429e27/Documentation/v2/errorcode.md

it shoud be a static string.

Reply:
can we throw the error code `401` at the same time?

Reply:
we checked the error code too.